pkg/libvmi/config.go

```diff
@@ -97,6 +109,19 @@ func WithDownwardAPIFs(name string) Option {
 	}
 }
 
+func newSysprepSecretVolume(secretName, volumeName string) v1.Volume {
+	return v1.Volume{
+		Name: volumeName,
+		VolumeSource: v1.VolumeSource{
+			Sysprep: &v1.SysprepSource{
+				Secret: &k8sv1.LocalObjectReference{
+					Name: secretName,
+				},
+			},
+		},
+	}
+}
+
```

Comment:
I think `newSysprepSecretVolume` and `newSysprepConfigMapVolume` can be merged together:
```

func newSysprepVolume(volumeName string, source *v1.SysprepSource) v1.Volume {
	return v1.Volume{
		Name: volumeName,
		VolumeSource: v1.VolumeSource{
			Sysprep: source,
		},
	}
}
```
and then simply pass the `SysprepSource`:
```
newSysprepVolume(volumeName, &v1.SysprepSource{
			ConfigMap: &k8sv1.LocalObjectReference{
				Name: configMapName,
			},
		}))
```